src/Forms/GridField/GridFieldPrintButton.php

```diff
@@ -228,21 +228,23 @@ public function generatePrintData(GridField $gridField)
 
         /** @var DataObject $item */
         foreach ($items->limit(null) as $item) {
-            $itemRow = new ArrayList();
+            if (!$item->hasMethod('canView') || $item->canView()) {
+                $itemRow = new ArrayList();
```

Comment:
since we're already touching this..
```suggestion
                $itemRow = ArrayList::create();
```